components/UI/Map/LeafletMap.tsx

```diff
@@ -189,13 +190,26 @@ function LeafletMap() {
         ]
       : DEFAULT_CENTER;
 
-  const defaultZoom = zoom
-    ? parseFloat(zoom as string)
-    : localCoordinatesURL
-    ? parseFloat(localCoordinatesURL.split("zoom=")[1].split("&")[0])
-    : device === "desktop"
-    ? DEFAULT_ZOOM
-    : DEFAULT_ZOOM_MOBILE;
+  useEffect(() => {
+    const zoomFromQuery = Number(zoom);
+    if (zoomFromQuery) {
+      setDefaultZoom(zoomFromQuery);
+      return;
+    }
+
+    const zoomParamFromURL = localCoordinatesURL
+      ?.split("zoom=")?.[1]
+      ?.split("&")?.[0];
+    if (zoomParamFromURL) {
+      setDefaultZoom(parseFloat(zoomParamFromURL));
+      return;
+    }
+
+    if (device == "mobile") {
+      setDefaultZoom(DEFAULT_ZOOM_MOBILE);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
let's create a custom hook for this: `useDefaultZoom`

Reply:
Done 🍻